pkg/keyspace/tso_keyspace_group.go

```diff
@@ -286,33 +290,70 @@ func (m *GroupManager) GetKeyspaceGroupByID(id uint32) (*endpoint.KeyspaceGroup,
 // DeleteKeyspaceGroupByID deletes the keyspace group by ID.
 func (m *GroupManager) DeleteKeyspaceGroupByID(id uint32) (*endpoint.KeyspaceGroup, error) {
 	var (
-		kg  *endpoint.KeyspaceGroup
-		err error
+		kg           *endpoint.KeyspaceGroup
+		err          error
+		moreToDelete = true
 	)
 
 	m.Lock()
 	defer m.Unlock()
-	if err := m.store.RunInTxn(m.ctx, func(txn kv.Txn) error {
-		kg, err = m.store.LoadKeyspaceGroup(txn, id)
+	for moreToDelete {
+		err = m.store.RunInTxn(m.ctx, func(txn kv.Txn) error {
+			kg, err = m.store.LoadKeyspaceGroup(txn, id)
+			if err != nil {
+				return err
+			}
+			if kg == nil {
+				return nil
+			}
+			if kg.IsSplitting() {
+				return ErrKeyspaceGroupInSplit
+			}
+
+			var (
+				keyspaceIDsToUnlock = make([]uint32, 0, maxTxnOps)
+				deleteKeyspaceCount = 0
+			)
+			defer func() {
+				for _, id := range keyspaceIDsToUnlock {
+					m.km.metaLock.Unlock(id)
+				}
+			}()
+			for _, id := range kg.Keyspaces {
+				if deleteKeyspaceCount >= maxTxnOps {
+					moreToDelete = true
```

Comment:
```suggestion
					moreToDelete = deleteKeyspaceCount < len(kg.Keyspaces)
```
Otherwise, if deleteKeyspaceCount == maxTxnOps, we still need to run one more unnecessary round.

Reply:
And if we don't use "moreToDelete = deleteKeyspaceCount < len(kg.Keyspaces)",  actually this assignment is no need because it must already be true here.